### PR TITLE
Alternative scrollview implementation (doesn't use "origin" for scroll offset)

### PR DESCRIPTION
**Pull request by nicoburns**

### CODE REVIEW

File: src/widget/core.rs [hunk 1]
```diff
@@ -498,6 +498,11 @@ impl Pod {
         }
     }
 
+    /// Get the widgets size (as returned by the layout method)
+    pub fn get_size(&mut self) -> Size {
```
Comment (Philipp-M) [inline src/widget/core.rs:502]: I still think we should omit `get_` prefixes to stay consistent :)
```suggestion
    pub fn size(&mut self) -> Size {
```

File: src/widget/scroll_view.rs [hunk 2]
```diff
@@ -95,28 +100,73 @@ impl Widget for ScrollView {
...
+            }
+        }
     }
 
-    fn layout(&mut self, cx: &mut LayoutCx, proposed_size: Size) -> Size {
-        let child_proposed = Size::new(proposed_size.width, 1e9);
-        let child_size = self.child.layout(cx, child_proposed);
-        Size::new(child_size.width, proposed_size.height)
+    fn layout(&mut self, cx: &mut LayoutCx, bc: &BoxConstraints) -> Size {
+        cx.request_paint();
```
Comment (Philipp-M) [inline src/widget/scroll_view.rs:129]: Is this necessary even, when the layout hasn't changed at all?

(I see this is done in other widgets as well though, maybe a point to keep in mind later for optimization etc.)

File: src/widget/scroll_view.rs [hunk 3]
```diff
@@ -95,28 +100,73 @@ impl Widget for ScrollView {
...
+
+        let cbc = BoxConstraints::new(
+            Size {
+                width: 0.0,
+                height: 0.0,
+            },
+            Size {
+                width: bc.max().width,
+                height: f64::INFINITY,
+            },
```
Comment (Philipp-M) [inline src/widget/scroll_view.rs:139]:
```suggestion
            Size::new(0.0, 0.0),
            Size::new(bc.max().width, f64::INFINITY),
```

(Nitpick, matter of preference I guess, doesn't really matter though)

File: src/widget/scroll_view.rs [hunk 4]
```diff
@@ -95,28 +100,73 @@ impl Widget for ScrollView {
...
+        );
+        let child_size = self.child.layout(cx, &cbc);
+        let size = Size {
+            width: child_size.width.min(bc.max().width),
+            height: child_size.height.min(bc.max().height),
+        };
```
Comment (Philipp-M) [inline src/widget/scroll_view.rs:145]:
```suggestion
        let size = Size::new(
            child_size.width.min(bc.max().width),
            child_size.height.min(bc.max().height),
        );
```

(same here)

**Review by Philipp-M — APPROVED** [review]
Apart from all the nitpicks, this LGTM, thanks :)

File: src/widget/core.rs [hunk 5]
```diff
@@ -498,6 +498,11 @@ impl Pod {
         }
     }
 
+    /// Get the widgets size (as returned by the layout method)
```
Comment (xStrom) [inline src/widget/core.rs:501]:
```suggestion
    /// Get the widget's size (as returned by the layout method)
```

**Review by xStrom — COMMENTED** [review]
Read through it and played around with the example, looks like a good step forward. 👍

I also like this more than #146, because it's closer to how Druid did it. The power of familiarity.

File: src/widget/core.rs (hunk 1 above)
Comment (nicoburns) [reply src/widget/core.rs]: I've updated this

File: src/widget/scroll_view.rs (hunk 2 above)
Comment (nicoburns) [reply src/widget/scroll_view.rs:130]: I'm not sure if this is the best way of doing it. But I was seeing the visual rendering on screen failing to update properly before I put this in.

**Review by xorgy — APPROVED** [review]
This is a good start, and doesn't regress anything.

**Pull request merged**